diff --git a/config/app.ts b/config/app.ts
index ffa2c22..7da939c 100644
--- a/config/app.ts
+++ b/config/app.ts
@@ -9,11 +9,15 @@ export const storeDir: string = process.env.STORE_DIR || joinPath(true, 'store')
 export const logsDir: string = joinPath(storeDir, 'logs')
 export const snippetsDir: string = joinPath(storeDir, 'snippets')
 
-export const tmpDir: string = joinPath(storeDir, '.tmp')
+export const tmpDir: string = process.env.TMPDIR || joinPath(storeDir, '.tmp')
 export const backupsDir: string = process.env.BACKUPS_DIR || joinPath(storeDir, 'backups')
 export const nvmBackupsDir: string = joinPath(backupsDir, 'nvm')
 export const storeBackupsDir: string = joinPath(backupsDir, 'store')
 
+export const settingsFile: string = process.env.SETTINGS_FILE || joinPath(storeDir, 'settings.json')
+export const scenesFile: string = process.env.SCENES_FILE || joinPath(storeDir, 'scenes.json')
+export const nodesFile: string = process.env.NODES_FILE || joinPath(storeDir, 'nodes.json')
+export const usersFile: string = process.env.USERS_FILE || joinPath(storeDir, 'users.json')
 
 export const defaultUser: string =  'admin'
 export const defaultPsw: string =  'zwave'
diff --git a/config/store.ts b/config/store.ts
index eb6ae24..b90cee4 100644
--- a/config/store.ts
+++ b/config/store.ts
@@ -1,5 +1,6 @@
 // config/store.js
 
+import { settingsFile, scenesFile, nodesFile, usersFile } from '../config/app';
 import { GatewayConfig } from '../lib/Gateway'
 import { MqttConfig } from '../lib/MqttClient'
 import { ZwaveConfig, deviceConfigPriorityDir } from '../lib/ZwaveClient'
@@ -25,7 +26,7 @@ export interface Settings {
 
 const store: Record<StoreKeys, StoreFile> = {
 	settings: {
-		file: 'settings.json',
+		file: settingsFile,
 		default: {
 			zwave: {
 				deviceConfigPriorityDir,
@@ -33,9 +34,9 @@ const store: Record<StoreKeys, StoreFile> = {
 			},
 		},
 	},
-	scenes: { file: 'scenes.json', default: [] },
-	nodes: { file: 'nodes.json', default: [] },
-	users: { file: 'users.json', default: [] },
+	scenes : { file: scenesFile, default: [] },
+	nodes : { file: nodesFile, default: [] },
+	users : { file: usersFile, default: [] }
 }
 
 export default store
diff --git a/lib/jsonStore.ts b/lib/jsonStore.ts
index 7e16d00..ce45cdc 100644
--- a/lib/jsonStore.ts
+++ b/lib/jsonStore.ts
@@ -94,7 +94,7 @@ export class StorageHelper {
 		let err: { code: string } | undefined
 		let data: any
 		try {
-			data = await readFile(utils.joinPath(storeDir, config.file))
+			data = await readFile(config.file)
 		} catch (error) {
 			err = error
 		}
@@ -127,7 +127,7 @@ export class StorageHelper {
 	}
 
 	async put(model: StoreFile, data: any) {
-		await writeFile(utils.joinPath(storeDir, model.file), data)
+		await writeFile(model.file, data)
 		this._store[model.file] = data
 		return data
 	}
